uaclient/contract.py

```diff
@@ -211,9 +211,21 @@ def request_updated_contract(
         )
     contract_client = UAContractClient(cfg)
     if contract_token:  # We are a mid ua-attach and need to get machinetoken
-        new_token = contract_client.request_contract_machine_attach(
-            contract_token=contract_token
-        )
+        try:
+            new_token = contract_client.request_contract_machine_attach(
+                contract_token=contract_token
+            )
+        except util.UrlError as e:
+            if isinstance(e, ContractAPIError):
+                if API_ERROR_INVALID_TOKEN in e:
+                    raise exceptions.UserFacingError(
+                        status.MESSAGE_ATTACH_FAILURE
+                    )
+            if "Temporary failure in name resolution" == str(e):
```

Comment:
This seems fragile to me. I wonder if we can make it more general. In what cases might we catch a `UrlError` (that isn't a `ContractAPIError`) where it wouldn't be a connectivity error?

Reply:
this may be a bigger topic than just the fragility here if we can confirm this works on trusty and xenial for the moment, maybe we can add a tech debt task to improve error handling

Reply:
Is this the message that you get when you unplug your system from the network?

Reply:
I'm cool with a generic connectivity error for now as a catch all and a tech debt item for more specific messages if and only if we have identified them.